src/Handlers/BaseRedisHandler.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace CodeIgniter\Settings\Handlers;
+
+use CodeIgniter\Cache\Handlers\RedisHandler;
+use Config\Cache;
+
+/**
+ * Redis class extension
+ */
+class BaseRedisHandler extends RedisHandler
```

Comment:
`Handlers\BaseRedisHandler` and `Handlers\RedisHandler`, at first, baffles me. Why are there two Redis handlers which do not inherit from the other? The `Base` prefix, and with `Abstract`, connotes that wherever it is affixed that is meant to be inherited from. But this is apparently not the case.

I'm not sure if the design is good.

Reply:
If we want to get rid of the class you could use a property reader. None of this is great practice but I totally get the intent. 
```php
$reader      = fn (): Redis => $this->redis;
$cache       = (new RedisHandler())->initialize();
$connection  = $reader->call($cache);
```

Reply:
It seems the use of this class is _just_ to retrieve the `Redis` instance. Why not use `Closure::bind()`?
```php
$cache = (new \CodeIgniter\Cache\Handlers\RedisHandler)->initialize();
$closure = Closure::bind(fn (): Redis => $this->redis, $cache, get_class($cache));

$closure();
```

Reply:
Why Settings depends on Cache?
It seems bad practice.

Reply:
> Why Settings depends on Cache? It seems bad practice.

The Cache dependency was just to retrieve the `Redis` instance.

Reply:
Settings should not depend on Cache.
If we need a Redis instance, we can create it with `new Redis()`.